server/src/uuid/model/taxonomy_term/mod.rs

```diff
@@ -236,3 +242,120 @@ impl TaxonomyTerm {
         typename.to_case(Case::Camel)
     }
 }
+
+#[derive(Error, Debug)]
+pub enum TaxonomyTermSetNameAndDescriptionError {
+    #[error(
+        "Taxonomy Term name and description cannot be set because of a database error: {inner:?}."
+    )]
+    DatabaseError { inner: sqlx::Error },
+    #[error(
+        "Taxonomy Term name and description cannot be set because of an event error: {inner:?}."
+    )]
+    EventError { inner: EventError },
+}
+
+impl From<sqlx::Error> for TaxonomyTermSetNameAndDescriptionError {
+    fn from(inner: sqlx::Error) -> Self {
+        Self::DatabaseError { inner }
+    }
+}
+
+impl From<EventError> for TaxonomyTermSetNameAndDescriptionError {
+    fn from(error: EventError) -> Self {
+        match error {
+            EventError::DatabaseError { inner } => inner.into(),
+            inner => Self::EventError { inner },
+        }
+    }
+}
+
+impl From<TaxonomyTermSetNameAndDescriptionError> for operation::Error {
+    fn from(error: TaxonomyTermSetNameAndDescriptionError) -> Self {
+        match error {
+            TaxonomyTermSetNameAndDescriptionError::DatabaseError {
+                inner: sqlx::Error::RowNotFound,
+            } => operation::Error::NotFoundError,
+            TaxonomyTermSetNameAndDescriptionError::DatabaseError { .. }
+            | TaxonomyTermSetNameAndDescriptionError::EventError { .. } => {
+                operation::Error::InternalServerError {
+                    error: Box::new(error),
+                }
+            }
+        }
+    }
+}
+
+impl TaxonomyTerm {
+    pub async fn set_name_and_description<'a, E>(
+        payload: &taxonomy_term_set_name_and_description_mutation::Payload,
+        executor: E,
+    ) -> Result<bool, TaxonomyTermSetNameAndDescriptionError>
+    where
+        E: Executor<'a>,
+    {
+        let mut transaction = executor.begin().await?;
+
+        let term_id = sqlx::query!(
+            r#"
+                SELECT term_id
+                    FROM term_taxonomy
+                    WHERE id = ?
+            "#,
+            payload.id
+        )
+        .fetch_one(&mut transaction)
```

Comment:
Idea: fetch_optional and map null to user input error

Reply:
good idea, unfortunately I couldn't do that in a clean way. Since in the api we would already do this check, we may leave it as it is. Otherwise, we could take a look together.